File: solution/java/day14/src/main/java/games/Result.java
package games;

public class Result<T> {
    private T success;

    public Result(T success) {
        this.success = success;
    }

    public Result() {
    }

    public static <T> Result<T> fromSuccess(T success) {
        return new Result<>(success);
    }

    public static <T> Result<T> failure() {
        return new Result<>();
    }

    public T success() {
        return success;
    }

    public boolean failed() {
        return success == null;
    }
}

File: solution/c#/Day14/Day14/Result.cs
namespace Day14
{
    public record Result<T>
    {
        private readonly T? _success;
        private Result(T success) => _success = success;

        private Result()
        {
        }

        public static Result<T> FromSuccess(T success) => new(success);
        public static Result<T> Failure() => new();

        public T? Success() => _success;
        public bool Failed() => _success == null;
    }
}